txpool/pool.go

```diff
@@ -624,8 +625,18 @@ func (p *TxPool) best(n uint16, txs *types.TxsRlp, tx kv.Tx, onTopOf, availableG
 
 			mt := best.ms[i]
 
-			if isYielding && mt.alreadyYielded {
-				continue
+			if len(toSkip) > 0 {
+				skip := false
+				for _, id := range toSkip {
```

Comment:
What about using something like github.com/deckarep/golang-set/v2 instead of slice?

Reply:
happy to add the dependency if you feel it would make the code easier to reason about?

Reply:
It probably doesn't matter performance-wise since the number of elements is relatively small in our use case, but IMO the set is a better fit here logically.